util/nodeReleaseData.ts

```diff
@@ -0,0 +1,35 @@
+import type { NodeReleaseStatus } from '../types/releases';
+
+export const getNodeReleaseStatus = (
```

Comment:
We can't use `NodeReleaseData` because it is called when we are building the `NodeReleaseData` itself.
Actually we can define a new data type but I prefer to keep it simple.

Reply:
Of course we can use the type, or at least a partial of the type.

Reply:
I believe having only one parameter for the release data is simpler than numerous arguments

Reply:
Something like this?
Or I think we can do type composition here (no need to flatten the `NodeReleaseData`)
```diff
diff --git a/hooks/useFetchNodeReleases.ts b/hooks/useFetchNodeReleases.ts
index 7d944beb..6b737365 100644
--- a/hooks/useFetchNodeReleases.ts
+++ b/hooks/useFetchNodeReleases.ts
@@ -2,7 +2,7 @@ import { useRouter } from 'next/router';
 import { useMemo } from 'react';
 import useSWR from 'swr';
 import { getNodeReleaseStatus } from '../util/nodeRelease';
-import type { NodeRelease } from '../types';
+import type { NodeRelease, NodeReleaseSupport } from '../types';

 const fetcher = (...args: Parameters<typeof fetch>) =>
   fetch(...args).then(res => res.json());
@@ -19,25 +19,23 @@ export const useFetchNodeReleases = (): NodeRelease[] => {
     const now = new Date();

     return data.map(raw => {
-      const status = getNodeReleaseStatus(
-        now,
-        raw.currentStart,
-        raw.ltsStart,
-        raw.maintenanceStart,
-        raw.endOfLife
-      );
+      const support = {
+        currentStart: raw.currentStart,
+        ltsStart: raw.ltsStart,
+        maintenanceStart: raw.maintenanceStart,
+        endOfLife: raw.endOfLife,
+      } as NodeReleaseSupport;
+
+      const status = getNodeReleaseStatus(now, support);

       return {
+        ...support,
         major: raw.major,
         version: raw.version,
         versionWithPrefix: `v${raw.version}`,
         codename: raw.codename || '',
         isLts: status === 'Active LTS' || status === 'Maintenance LTS',
         status: status,
-        currentStart: raw.currentStart,
-        ltsStart: raw.ltsStart,
-        maintenanceStart: raw.maintenanceStart,
-        endOfLife: raw.endOfLife,
         npm: raw.npm || '',
         v8: raw.v8 || '',
         releaseDate: raw.releaseDate || '',
diff --git a/types/releases.ts b/types/releases.ts
index 346fc072..1b03e22d 100644
--- a/types/releases.ts
+++ b/types/releases.ts
@@ -16,7 +16,7 @@ export type NodeReleaseStatus =
   | 'End-of-life'
   | 'Pending';

-export interface NodeRelease {
+export type NodeRelease = {
   major: number;
   version: string;
   versionWithPrefix: string;
@@ -31,4 +31,9 @@ export interface NodeRelease {
   v8: string;
   releaseDate: string;
   modules: string;
-}
+};
+
+export type NodeReleaseSupport = Pick<
+  NodeRelease,
+  'currentStart' | 'ltsStart' | 'maintenanceStart' | 'endOfLife'
+>;
diff --git a/util/nodeRelease.ts b/util/nodeRelease.ts
index c503cfb3..d9249fd2 100644
--- a/util/nodeRelease.ts
+++ b/util/nodeRelease.ts
@@ -1,32 +1,29 @@
-import type { NodeReleaseStatus } from '../types/releases';
+import type { NodeReleaseStatus, NodeReleaseSupport } from '../types/releases';

 export const getNodeReleaseStatus = (
   now: Date,
-  currentStart?: string,
-  ltsStart?: string,
-  maintenanceStart?: string,
-  endOfLife?: string
+  support: NodeReleaseSupport
 ): NodeReleaseStatus => {
-  if (endOfLife) {
-    if (now > new Date(endOfLife)) {
+  if (support.endOfLife) {
+    if (now > new Date(support.endOfLife)) {
       return 'End-of-life';
     }
   }

-  if (maintenanceStart) {
-    if (now > new Date(maintenanceStart)) {
+  if (support.maintenanceStart) {
+    if (now > new Date(support.maintenanceStart)) {
       return 'Maintenance LTS';
     }
   }

-  if (ltsStart) {
-    if (now > new Date(ltsStart)) {
+  if (support.ltsStart) {
+    if (now > new Date(support.ltsStart)) {
       return 'Active LTS';
     }
   }

-  if (currentStart) {
-    if (now >= new Date(currentStart)) {
+  if (support.currentStart) {
+    if (now >= new Date(support.currentStart)) {
       return 'Current';
     }
   }
```

Reply:
For me it's hard to visualise this GIST, but I already approved your code, so sounds good!